### ISSUE TITLE
Firefox 47 breaks tabkit2

### ISSUE THREAD

**Issue by Ligerx** [tabkit/tabkit2#117]
So Firefox started acting really weird today, and I believe it's because it upgraded to the latest version. Hopefully this is descriptive, because there are some very weird side effects.

- It seems like hover and/or click only works on a part of the bottom screen.
- Right click opens the menu correctly, but clicking options does nothing. For example, clicking new tab does not work.
- I can drag the window around by clicking and holding inside of a webpage where it should normally just highlight html.
- Clicking on things like links only works if I double click them. However, this causes my window to resize and I think it triggers 2 clicks.

**Comment by Perfido** [comment]
+1 (same behavior)
It's a major issue

**Comment by krobe** [comment]
I am seeing the same behavior with FF47 x86 on Windows 10 x64. 25 tabs open.

**Comment by eriknw26** [comment]
Same trouble as soon as updated to FF47 x86 on Windows 10 x64. No links or buttons respond unless double-clicked, which causes the window to resize at the same time.

**Comment by Perfido** [comment]
Some other things :
- I add 15 tabs hidden under their parent tab when update occurs, but their closing had no effet
- I have 4 groups of tabs with 14, 11 and less tabs
- behavior appends on other FF window with only 1 tab and "main" FF window and it's 45 tabs in background, but then "link disabled" area can be different (90%) from "main" window (65%) AND this area is different in full screen or windowed mode AND this area is more important if personnal bar is hidden. In windowed mode, it could variing (65% stable then 90% stable with same window after changing things) oO
- I haven't try to close all tabs and test with a "full clean" FF

**Comment by PikachuEXE** [comment]
Thanks all for reporting
It will be fixed when closed to next ESR version release

**Comment by industriflow** [comment]
Confirming:  can't click URLs in Firefox ver. 47 with TabKit2 ver. 0.12.14

Confirmed with no other extensions active, except TK2. (Windows 7 Enterprise 32-bit w/ all updates).
Problem occurs with even one tab open.

Can middle-click a URL, which in some configurations will open the URL in a new tab.
Can right-click to get a drop-down menu.

When is the next Firefox ESR scheduled?

**Comment by Ligerx** [comment]
@industriflow according to the release schedule, on 8/2

[https://wiki.mozilla.org/RapidRelease/Calendar](https://wiki.mozilla.org/RapidRelease/Calendar)

**Comment by amatesi** [comment]
let's show some support for PikachuEXE - let's donate here ("Contribute"-button): https://addons.mozilla.org/en-US/firefox/addon/tab-kit-tab-highlighter/

**Comment by PikachuEXE** [comment]
The main reason for supporting ESR only is lack of time

Also I am quite worried that the same issue will be fixed/reappear in later versions
The Firefox team now does NOT care about compatibility with addons (starting from version 4)
It surely makes thing broken more often but at least it can still be worked around by using ESR

So I strongly recommend you to use ESR versions which still has security updates
And you will have less issues with addons (at least until you update to next ESR version, slight less than a year)

**Comment by amatesi** [comment]
Hi PikachuEXE and thank you for your clarification.

FWIW, I proceeded as follows:
1. Uninstalled Firefox 47.0 & Rebooted as requested.
2. Installed Firefox ESR 45.
3. Re-enabled tabkit.

NOTE: the process RETAINED all my open tabs and settings (ie. I didn't have to reconfigure everything from scratch).

All good for me/Tabkit 2 reported as working on Firefox ESR 45.

Cheers.

**Comment by DShillito** [comment]
I wish I'd read this before upgrading :(
I didn't realise you only supported the ESR version (TBH I did not know what ESR versions were until this).
Not that I blame you - this is Mozilla's fault for not respecting its community of developers.
I will try downgrading.

**Comment by DShillito** [comment]
Good point @amatesi. Tab Kit 2nd Edition is the first plugin I install on a fresh Firefox.
I really should show my appreciation.

**Comment by labyrinth-to-solve** [comment]
Because of this bug I still run Ffox 44.0.2. Is there any chance this will be solved in realistic future or do I have to search for another solution? For me the Tabkit2 is one of the most important tools.

**Comment by M3Rocket** [comment]
@labyrinth-to-solve: You should at least upgrade to [FF ESR 45.3](https://goo.gl/pRZVlL)--TabKit2 will run without problems. No worries--when you install the ESR version, it will automatically copy over your old settings from FF 44.0.2.

**Comment by DShillito** [comment]
I'm running 46.0.1 and it works fine for me, sadly it does not work with the new 48.0 :(

I'm tempted to download the source and poke around to see if there is any obvious fix for the issues that affect me (mainly right clicking links and opening in a new tab), however I've never done any Mozilla addin dev, so it could take me some time.

**Comment by PikachuEXE** [comment]
The reason I am not fixing this issue yet is:
It might be broken again 1 version after.
So I fallback to support ESR only. At least the extension works fine for about a year.

**Comment by labyrinth-to-solve** [comment]
Good enough. I will switch to ESR. Thanks for a great job!

**Comment by RC-3** [comment]
SOLUTION: 
I noticed I had this issue on one computer (~200 tabs open) but not on another computer (< 30 tabs) which I thought was weird. I managed to fix it by refreshing the Classic Theme Restorer settings.

I fixed it on my affected computer by 
1) Tools > Classic Theme Restorer > Settings > Restore Defaults
2) Reboot FF
3) Tabs > Squared Tabs (Classic)
4) Tabs not on top (v2)
And it started working again!

I don't now if this will help others. I also don't know if this just means refreshing Classic Tab settings clears out whatever causes it?
Give it a shot and let us know if it works for you.

@PikachuEXE I LOVE YOUR EXTENSION!! Thank you so much for all your hard work. :D

**Comment by sheandherwork** [comment]
I can confir RC-3's solution works! Big THANK YOU!
